Mathlib/AlgebraicGeometry/EllipticCurve/Affine.lean

```diff
@@ -656,6 +685,12 @@ lemma zero_def : (zero : W.Point) = 0 :=
   rfl
 #align weierstrass_curve.point.zero_def WeierstrassCurve.Affine.Point.zero_def
 
+lemma some_ne_zero {x y : R} (h : W.Nonsingular x y) : some h ≠ 0 := by rintro (_|_)
+
+lemma some_eq_some_iff {x₁ x₂ y₁ y₂ : R} (h₁ : W.Nonsingular x₁ y₁) (h₂ : W.Nonsingular x₂ y₂) :
```

Comment:
Can we not get `some_eq_some_iff` from `ext` (as a tag in the structure or however else?)?

Reply:
`@[ext]` before the definition `inductive Point` produces the error
```
@[ext] attribute only applies to structures or theorems proving x = y, got Type u
```
so I guess it's not yet supported.
I seem to recall that there's some `mk_inj` stuff (in Lean 3?) but I can't find it now.